lib/routes/charge/pos_invoice.dart

```diff
@@ -396,20 +396,22 @@ class POSInvoiceState extends State<POSInvoice> {
       }
 
       bool flipFiat = _useFiat == (currency != null);
+      Int64 oldSatAmount = _satAmount(accountModel, amount);
       if (flipFiat) {
         _useFiat = !_useFiat;
-        _clearAmounts(clearTotal: true);
+        _clearAmounts();
       }
+      amount = (flipFiat && _useFiat || _useFiat) // For Sat->Fiat or Fiat->Fiat
```

Comment:
I suggest to replace with this:

```
amount = oldSatAmount.toDouble();

// We need to convert only in case we use fiat.
if (_useFiat) {
    amount = accountModel.fiatCurrency.satToFiat(oldSatAmount);
}
```

Reply:
[Done](https://github.com/breez/breezmobile/pull/282/commits/28d751d7fb4f91cac5ddca6bd005a3214879aab6)
I kept `accountModel.fiatConversionList.firstWhere((f) => f.currencyData.shortName == value)` as accountModel.fiatCurrency is not updated till we reach here.